src/Microsoft.AspNetCore.Authentication.Impl/AuthenticationServiceCollectionExtensions.cs

```diff
@@ -46,8 +45,7 @@ public static IServiceCollection AddAuthentication(this IServiceCollection servi
             return services;
         }
 
-        // REVIEW: rename to just AddScheme?
-        public static IServiceCollection AddSchemeHandler<TOptions, THandler>(this IServiceCollection services, string authenticationScheme, Action<TOptions> configureOptions)
+        public static IServiceCollection AddScheme<TOptions, THandler>(this IServiceCollection services, string authenticationScheme, Action<TOptions> configureOptions, bool canHandleRequests)
```

Comment:
> Looking the ASOS, you would just need to add them all as callback events, and should be good to go I think.

You mean as callback path? In this case, this won't work. The `MatchEndpoint` event MUST be called for every request to allow the developer to decide whether it's a request that should be handled or not, depending on his own policy (not just the path).

Not to mention that I ultimately want to support URLs like /connect/register/[dynamic routing parameter]).